docs/DesignDocs/SealingAPI.md

```diff
@@ -0,0 +1,591 @@
+`oe_seal()`? Here are 3 options:
+
+1. `oe_seal()` takes the seal key directly as a function argument; or
+2. `oe_seal()` takes a `key_info` as its argument and derives the desired seal
+   key; or
+3. `oe_seal()` takes a `seal_policy` as its argument, and calls
+   `oe_get_seal_key_by_policy()` to obtain the seal key.
+
+Option 1 implies access to the seal key by software, which isn't always true
+(e.g., TPM never reveals seal keys to software). Option 3 prohibits developers
+from customing `key_info` (probably in a TEE-specific manner), which violates
+the *Accommodative* requirement. Option 2 is our choice.
+
+Now that `key_info` has been passed to `oe_seal()` and it has to be persisted
+for `oe_unseal()`, it's a natural idea to embed `key_info` into the sealed
+blob. Combined with the considerations above we come to declarations of
+`oe_seal()` and `oe_unseal()` that are shown below. Meanings of the function
+parameters should be intuitive. More details can be found in
+[oe_seal()](#oe_seal()) and [oe_unseal()](#oe_unseal()) later in this document.
+
+```C
+oe_result_t oe_seal(
+    const uint8_t* key_info,
+    size_t key_info_size,
+    const uint8_t* plaintext,
+    size_t plaintext_size,
+    const uint8_t* additional_data,
+    size_t additional_data_size,
+    uint8_t **blob,
+    size_t *blob_size);
+
+oe_result_t oe_unseal(
+    uint8_t* blob,
+    size_t blob_size,
+    uint8_t** plaintext,
+    size_t* plaintext_size,
+    uint8_t** additional_data,
+    size_t* additional_data_size);
+```
+
+The code snippet below shows how to encrypt data (i.e., `my_data_size` bytes
+starting at `my_data`) using `oe_seal()`.
+
+```C
+uint8_t* blob;
+size_t blob_size;
+result = oe_seal(key_info, key_info_size,
+                 my_data,       // plaintext
+                 my_data_size,  // plaintext_size
+                 NULL,          // additional_data
+                 0,             // additional_data_size
+                 &blob, &blob_size);
+id (result != OE_OK)
+    // Handle encryption error...
+else
+{
+    // Persist blob...
+
+    oe_free(blob); // Free blob when done
+}
+```
+
+Please note that `my_data` above will be both encrypted and
+integrity-protected. But if only integrity needs protection, one could instead
+pass `my_data` and `my_data_size` as `additional_data` and
+`additional_data_size`, respectively.
+
+```C
+uint8_t* blob;
+size_t blob_size;
+result = oe_seal(key_info, key_info_size,
+                 NULL,          // plaintext
+                 0,             // plaintext_size
+                 my_data,       // additional_data
+                 my_data_size,  // additional_data_size
+                 &blob, &blob_size);
+id (result != OE_OK)
+    // Handle encryption error...
+else
+{
+    // Persist blob...
+
+    oe_free(blob); // Free blob when done
+}
+```
+
+Please note that the blob format is proprietary so there's no architectural way
```

Comment:
*Proprietary* here means the blob format is subject to change, because the cipher and mode may change.  So the users shouldn't assume the internal format of the sealed blob. This I think is a common practice, as TPM spec isn't specific on the cipher/mode and sealed blob format either.

Reply:
then change "proprietary" to "internal"

Reply:
```suggestion
Please note that the blob format is internal to the OE SDK so there's no architectural way
```